plugins/aws/rds/idleRDSInstance.js

```diff
@@ -0,0 +1,132 @@
+        var rds_idle_instance_writeIOPS_percentage = settings.rds_idle_instance_writeIOPS_percentage || this.settings.rds_idle_instance_writeIOPS_percentage.default;
+        rds_idle_instance_cpu_percentage = parseFloat(rds_idle_instance_cpu_percentage);
+
+        async.each(regions.rds, function(region, rcb) {
+            var describeDBInstances = helpers.addSource(cache, source,
+                ['rds', 'describeDBInstances', region]);
+
+            if (!describeDBInstances) return rcb();
+
+            if (describeDBInstances.err || !describeDBInstances.data) {
+                helpers.addResult(
+                    results, 3,
+                    `Unable to query for RDS instance: ${helpers.addError(describeDBInstances)}`, region);
+                return rcb();
+            }
+
+            if (!describeDBInstances.data.length){
+                helpers.addResult(results, 0, 'No RDS instance found', region);
+                return rcb();
+            }
+
+            describeDBInstances.data.forEach(instance => {
+                if (!instance.DBInstanceArn) return;
+           
+                var getRdsMetricStatistics = helpers.addSource(cache, source,
+                    ['cloudwatch', 'getRdsMetricStatistics', region, instance.DBInstanceIdentifier]);
+                var getRdsReadIOPSMetricStatistics = helpers.addSource(cache, source,
+                    ['cloudwatch', 'getRdsReadIOPSMetricStatistics', region, instance.DBInstanceIdentifier]);
+                var getRdsWriteIOPSMetricStatistics = helpers.addSource(cache, source,
+                    ['cloudwatch', 'getRdsWriteIOPSMetricStatistics', region, instance.DBInstanceIdentifier]);
+
+                if (!getRdsMetricStatistics || getRdsMetricStatistics.err ||
+                    !getRdsMetricStatistics.data || !getRdsMetricStatistics.data.Datapoints) {
+                    helpers.addResult(results, 3,`Unable to query for CPU metric statistics: ${helpers.addError(getRdsMetricStatistics)}`, region, instance.DBInstanceArn);
+                    return;
+                }
+
+                if (!getRdsMetricStatistics.data.Datapoints.length) {
+                    helpers.addResult(results, 0,'CPU metric statistics are not available', region, instance.DBInstanceArn);
+                    return;
+                }
+                   
+                   
+                if (!getRdsReadIOPSMetricStatistics || getRdsReadIOPSMetricStatistics.err ||
+                        !getRdsReadIOPSMetricStatistics.data || !getRdsReadIOPSMetricStatistics.data.Datapoints) {
+                    helpers.addResult(results, 3, `Unable to query for Read IOPS metric statistics: ${helpers.addError(getRdsReadIOPSMetricStatistics)}`, region, instance.DBInstanceArn);
+                    return;
+                }
+    
+                if (!getRdsReadIOPSMetricStatistics.data.Datapoints.length) {
```

Comment:
Even if they are not available, we can check other datapoints. We should return only when all 3 datapoints are not available.